### PR TITLE
Move stored flag from TextSearchInfo to MappedFieldType

### PR DESCRIPTION
**Pull request by javanna**
Move stored flag from TextSearchInfo to MappedFieldType

### CODE REVIEW

**Comment by javanna** [comment]
run elasticsearch-ci/packaging-sample-windows

File: server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java [hunk 1]
```diff
@@ -105,7 +104,7 @@ public SourceFieldMapper build(BuilderContext context) {
         public static final SourceFieldType INSTANCE = new SourceFieldType();
 
         private SourceFieldType() {
-            super(NAME, false, false, TextSearchInfo.NONE, Collections.emptyMap());
+            super(NAME, false, false, true, TextSearchInfo.NONE, Collections.emptyMap());
```
Comment (romseygeek) [inline server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java:107]: I think we need to pass `false` here if `enabled` is `false`?

**Review by romseygeek — APPROVED** [review]
LGTM, thanks @javanna!

I have one question about source metadata fields, but no need for another go-round.

File: server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java (hunk 1 above)
Comment (javanna) [reply server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java:107]: yes I meant to get back to this, I was not sure either. I looked at it from the perspective of "if enabled there is no _source", but actually _source is still mapped and stored should be set to false. will update.

Comment (javanna) [reply server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java:107]: and btw, I am setting doc_values to true rather than stored :D

**Pull request merged**